sql/mozfun/iap/derive_apple_subscription_interval/udf.sql

```diff
@@ -1,119 +1,96 @@
-  ),
-  assert.equals(
-    "2 day",
-    iap.derive_apple_subscription_interval(
-      DATETIME(TIMESTAMP "2021-03-15 01:00:00 UTC", "America/Los_Angeles"),
-      DATETIME(TIMESTAMP "2021-03-17 01:00:00 UTC", "America/Los_Angeles")
-    )
-  ),
-  -- daylight savings time
-  assert.equals(
-    "quarter",
-    iap.derive_apple_subscription_interval(
-      DATETIME(TIMESTAMP "2021-03-14 01:00:00 UTC", "America/Los_Angeles"),
-      DATETIME(TIMESTAMP "2021-06-14 00:00:00 UTC", "America/Los_Angeles")
-    )
-  ),
-  assert.equals(
-    "month",
-    iap.derive_apple_subscription_interval(
-      DATETIME(TIMESTAMP "2021-03-14 01:00:00 UTC", "America/Los_Angeles"),
-      DATETIME(TIMESTAMP "2021-04-14 00:00:00 UTC", "America/Los_Angeles")
-    )
-  ),
-  assert.equals(
-    "week",
-    iap.derive_apple_subscription_interval(
-      DATETIME(TIMESTAMP "2021-03-14 01:00:00 UTC", "America/Los_Angeles"),
-      DATETIME(TIMESTAMP "2021-03-21 00:00:00 UTC", "America/Los_Angeles")
-    )
-  ),
-  assert.equals(
-    "day",
-    iap.derive_apple_subscription_interval(
-      DATETIME(TIMESTAMP "2021-03-14 01:00:00 UTC", "America/Los_Angeles"),
-      DATETIME(TIMESTAMP "2021-03-15 00:00:00 UTC", "America/Los_Angeles")
-    )
-  ),
-  assert.equals(
-    "2 day",
-    iap.derive_apple_subscription_interval(
-      DATETIME(TIMESTAMP "2021-03-14 01:00:00 UTC", "America/Los_Angeles"),
-      DATETIME(TIMESTAMP "2021-03-16 00:00:00 UTC", "America/Los_Angeles")
-    )
-  ),
-  -- february vs march PST
-  assert.equals(
-    "month",
-    iap.derive_apple_subscription_interval(
-      DATETIME(TIMESTAMP "2021-03-01 01:00:00 UTC", "America/Los_Angeles"),
-      DATETIME(TIMESTAMP "2021-03-29 00:00:00 UTC", "America/Los_Angeles")
-    )
-  ),
+  assert.equals(expected, iap.derive_apple_subscription_interval(start, `end`))
+FROM
+  UNNEST(
+    ARRAY<
+      STRUCT<
+        expected STRUCT<`interval` STRING, interval_count INT64>,
+        start DATETIME,
+        `end` DATETIME
+      >
+    >[
+      (("year", 1), DATETIME "2020-03-15 00:00:00", DATETIME "2021-03-15 00:00:00"),
+      (("month", 6), DATETIME "2021-03-15 00:00:00", DATETIME "2021-09-15 00:00:00"),
+      (("month", 3), DATETIME "2021-03-15 00:00:00", DATETIME "2021-06-15 00:00:00"),
+      (("month", 1), DATETIME "2021-03-15 00:00:00", DATETIME "2021-04-15 00:00:00"),
+      (("week", 2), DATETIME "2021-03-15 00:00:00", DATETIME "2021-03-29 00:00:00"),
+      (("week", 1), DATETIME "2021-03-15 00:00:00", DATETIME "2021-03-22 00:00:00"),
+      (("day", 2), DATETIME "2021-03-15 00:00:00", DATETIME "2021-03-17 00:00:00"),
+      (("day", 1), DATETIME "2021-03-15 00:00:00", DATETIME "2021-03-16 00:00:00"),
+      -- last day of month
+      (("month", 1), DATETIME "2021-02-28 00:00:00", DATETIME "2021-03-31 00:00:00"),
+      (("month", 1), DATETIME "2021-01-31 00:00:00", DATETIME "2021-02-28 00:00:00"),
+      (("month", 1), DATETIME "2021-02-28 00:00:00", DATETIME "2021-03-28 00:00:00"),
```

Comment:
If the latter date were "2021-03-27", would this be ("month", 0)?

Reply:
that would come out to `("day", 27)`